Week 2/Extra Practice Problems (Week 2)/2597: The Number of Beautiful Subsets/dominic.cpp

```diff
@@ -0,0 +1,28 @@
+typedef pair<int, int> pii;
+
+class Solution {
+public:
+    int beautifulSubsets(vector<int>& nums, int k) {
+        int n = nums.size();
+        
+        vector<pii> d;
+        for (int i = 0; i < n; i++) {
+            for (int j = i+1; j < n; j++) {
+                if (nums[i]+k != nums[j] && nums[i]-k != nums[j]) continue;
+                d.emplace_back(1<<i, 1<<j);
+            }
+        }
+        
+        int count = 0;
+        for (int i = 0; i < (1<<n); i++) {
+            for (auto &j : d) {
```

Comment:
Technically we don't have to check all pairs here. Though since `n` is small, it doesn't matter.

Alternatively, let's use `i` as the mask for the current subset, and check if the subset is valid?